ci/scripts/check-release-versions.js

```diff
@@ -36,11 +37,122 @@ function checkReleaseVersion (containerName, tagToBeDeployed, currentContainerDe
   const currentRelease = Number(currentImage.split(':')[1].split('-')[0])
   const releaseToBeDeployed = Number(tagToBeDeployed.split('-')[0])
   if (releaseToBeDeployed < currentRelease) {
-    throw new Error(`
-        You are trying to deploy release  ${releaseToBeDeployed} of ${containerName}
-        which is older than the current release ${currentRelease}. Bailing out.
-        If you need to deploy this version please deploy manually using terraform.`)
+    console.log(
+      `You are trying to deploy release  ${releaseToBeDeployed} of ${containerName} ` +
+      `which is older than the current release ${currentRelease}.\n\n` +
+
+      `Checking that ${releaseToBeDeployed} is the latest ENABLED release.`)
+
+    const resourceName = ecrResourceName(containerName)
+    const latestEnabledRelease = await getLatestEnabledRelease(resourceName)
+    console.log(`The latest ENABLED release is ${latestEnabledRelease}`)
+    if (tagToBeDeployed < latestEnabledRelease) {
+      throw new Error(
+      `${releaseToBeDeployed} of ${containerName} is not the latest ENABLED version. Bailing out.\n` +
+      'If you need to deploy this version you must disable later versions (pinning will not work)')
+    }
+    console.log(`Proceeding to deploy release ${releaseToBeDeployed}`)
+  }
+}
+
+function ecrResourceName (containerName) {
+  if (containerName === 'nginx') {
+    containerName += '-proxy'
+  }
+
+  const { PIPELINE_NAME, CONTAINER_SUFFIX } = process.env
+
+  if (!PIPELINE_NAME) {
+    throw new Error('PIPELINE_NAME env var not set. Cannot query concourse for resource versions')
+  }
+  if (!CONTAINER_SUFFIX) {
+    throw new Error('CONTAINER_SUFFIX env var not set. Cannot query concourse for resource versions')
+  }
+
+  return `${PIPELINE_NAME}/${containerName}-ecr-registry-${CONTAINER_SUFFIX}`
+}
+
+async function getLatestEnabledRelease (resource) {
+  const flyTeam = await flyLogin()
+
+  let tryCount = 0
+  const MAX_ATTEMPTS = 5
+  const RETRY_WAIT_TIME_SECONDS = 5
+  let latestEnabledRelease
+
+  do {
+    tryCount += 1
+    const releases = await getReleases(resource, flyTeam)
+    latestEnabledRelease = releases.find((release) => release.enabled)
+
+    if (!latestEnabledRelease) {
+      console.log(`There are no enabled releases for ${resource}, sometimes this is due to flakiness in the fly api call`)
+      console.log(`Retrying, attempt ${tryCount} of ${MAX_ATTEMPTS} in ${RETRY_WAIT_TIME_SECONDS} seconds`)
+      await new Promise(resolve => setTimeout(resolve, RETRY_WAIT_TIME_SECONDS * 1000))
```

Comment:
There's a nicer way to sleep these days

```
const { setTimeout } = require('node:timers/promises');

await setTimeout(1000)
```

If you need setTimeout and the promise version you can

```
const { setTimeout: setTimeoutPromise } = require('node:timers/promises');

await setTimeoutPromise(1000)
```